### ISSUE TITLE
Possible issue with PCR "^\\section"

### ISSUE THREAD

**Issue by priyadarshan** [joddie/pcre2el#26]
Hi,
`(rxt-pcre-to-elisp "^\\section")` gives
```
"^[	
^l^M ]ection"
```

Is that the correct behaviour?

**Comment by priyadarshan** [comment]
I have also this:
```
(rxt-pcre-to-rx "\d+")
;; (+ "^?")
```
Could it be something to do tih Emacs 25 (which I am using)?

**Comment by priyadarshan** [comment]
Interestingly, using `pcre2el` as backend to `visual-regexp-steroids`, the PCRE syntax seems to work (see image)
![Uploading screenshot 2016-04-08 at 13.54.03.png…]()

**Comment by priyadarshan** [comment]
I have tested this issue with Emacs 24.1 on a blank .emacs (beside loading `pcre2el`) and I have the same result, ie `(rxt-pcre-to-elisp "^\\section")` returns
```
"^[	
^L^M ]ection"
```

Perhaps I am confusing elisp regex, or probably not using correctly that function?

**Comment by priyadarshan** [comment]
Reading carefully all the examples, I realised that one needs to double up the backslashes.
I knew that was the case with elisp, but I thought `pcre2el` was meant to preserve exact perl syntax?

The example regex would be working as 
```
 (re-search-forward (rxt-pcre-to-elisp "^\\\\section") nil t)
```

That is fine of course, but it means one could not use Perl RE as-is, sharing perhaps a common repository of filter, but changed them before using them.

Forgive my ignorance in the subject, would not be possible to fix that using a macro?

**Comment by vsrepo** [comment]
I was also wondering about this. Interestingly, `re-builder` with `pcre` syntax works with PCRE like `"^\\section"`, or `(\d+)`.
It would be quite useful to have the same functionality, ie, being able to enter real PCRE regex like rebuilder+pcre above, but programmatically, for example `(rxt-pcre-to-elisp "(\d+)"`

**Comment by zw963** [comment]
@priyadarshan ,  This should be the correct behavior we want.
"\\s" in emacs strings is explained to "\s", which match any blank
char (\r, \t, \s) in perl,  what you see `^[ 
^l^M ]' just the literal form of those character.

if you want match `\s` literal form, you have to told it this is literal 
form,  escape the first `\` with `\\`, the result is: `\\s`.

So, you need feed `rxt-pcre-to-elisp` with `"\\\\s"`, what `rxt-pcre-to-elisp`
see is just `\\s`.

I think this is hard to resolve like ruby/perl, unless elisp support a new literal
presentation like `/regexp/` instead of `"regexp"`.

**Comment by zw963** [comment]
> I need to specify (rxt-pcre-to-elisp "(\\d+)") which is not a real PCRE anymore.

The same course(see previous reply), if we need a real PCRE, we need a new
 presentation which can skip emacs lisp literal strings representation. I think.

**Comment by zw963** [comment]
Oh, @joddie (the author) seem like still on travelling now. 
I have another issue not resolved until now.
Still need wait ...

**Comment by priyadarshan** [comment]
> The same course(see previous reply), if we need a real PCRE, we need a new
> presentation which can skip emacs lisp literal strings representation. I think.

Yes, that's what I thought. It would be quite useful, at least to some, to have a way to add real PCRE.

My use case is having to use hundreds of evolving regexes from Perl and Php scripts upstream, and the need to use them with elisp.

To be able to feed them as they are to `pcre2el` would be a boon indeed. One could maintain a regex corpus only in one form.
